### ISSUE TITLE
Pmag_gui is not starting 

### ISSUE THREAD

**Issue by hl04445567** [PmagPy/PmagPy#589]
**What command was run:**
Opening Pmag_gui windows standalone client

**What happened:**
Error message says "Failed to execute script pmag_gui"

Below is the message from wxPython terminal:

-I- PmagPy icon file not found -- skipping
-I- Read in any available data from working directory
-I- Using online data model
-I- Couldn't connect to earthref.org, using cached method codes
-I- Using cached method codes
-I- Using cached vocabularies
Traceback (most recent call last):
  File "pmagpy\contribution_builder.py", line 48, in __init__
AttributeError: type object 'Contribution' has no attribute 'vocab'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "site-packages\pandas\core\indexes\base.py", line 2657, in get_loc
  File "pandas/_libs/index.pyx", line 108, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 132, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 1601, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1608, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'derived_value'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "programs\pmag_gui.py", line 871, in <module>
  File "programs\pmag_gui.py", line 860, in main
  File "programs\pmag_gui.py", line 81, in __init__
  File "programs\pmag_gui.py", line 349, in get_dir
  File "programs\pmag_gui.py", line 378, in on_change_dir_button
  File "programs\pmag_gui.py", line 393, in on_finish_change_dir
  File "programs\pmag_gui.py", line 139, in get_wd_data
  File "pmagpy\contribution_builder.py", line 50, in __init__
  File "pmagpy\controlled_vocabularies3.py", line 46, in __init__
  File "pmagpy\controlled_vocabularies3.py", line 379, in get_all_vocabulary
  File "pmagpy\controlled_vocabularies3.py", line 258, in get_controlled_vocabularies
  File "site-packages\pandas\core\frame.py", line 2927, in __getitem__
  File "site-packages\pandas\core\indexes\base.py", line 2659, in get_loc
  File "pandas/_libs/index.pyx", line 108, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 132, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 1601, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1608, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'derived_value'

**What I expected to happen:**
Pmag_gui normally opens

P.S. I've been using Pmag_gui for a while, this problem started to occur a few days ago. I redownloaded the Windows standalone version, didn't work.
**Data file(s):**

**Screenshot(s):**
![image](https://user-images.githubusercontent.com/12456638/61919878-1f7acd00-af8a-11e9-8f87-6d2ed28163aa.png)

**PmagPy version:**
pmag_gui_4.2.23

**Comment by njarboe** [comment]
This is likely a problem with the data file and not pmag_gui. I have just downloaded pmag_gui for windows and it ran with a known good datafile. If you would upload the datafile(s) you are trying to view, then I can test them out and take a look at them for correctness.
Here is a screenshot of my version running:
![Screen Shot 2019-07-27 at 2 28 02 PM](https://user-images.githubusercontent.com/1278789/61999664-4238de80-b07b-11e9-940a-f460fc01b9eb.png)

**Comment by hl04445567** [comment]
Thanks for the replying.

I didn't think it was because of my dataset, mainly because that I have mutiple datasets, which I've done PCA fittings normally in pmag_gui before. I cannot open none of them now. I think it's unlikly that they all go wrong at the same time?

From your wxPython window, I noticed that one line says Getting method codes from earthref.org etc., while in the same place mine window shows 

-I- Couldn't connect to earthref.org, using cached method codes
-I- Using cached method codes
-I- Using cached vocabularies
Traceback (most recent call last):
File "pmagpy\contribution_builder.py", line 48, in init
AttributeError: type object 'Contribution' has no attribute 'vocab'

Could this have something to do with the errors I got?

**Comment by njarboe** [comment]
Is your computer connected to the internet when you are trying to run pmag_gui? If not, give that a try and it should be able to download file from earthref.org.

If you don't mind uploading one of your datasets that is not working, that could be very useful. I could try opening it and see if it works on my setup and use it for finding out more of where the problem might be.

**Comment by ltauxe** [comment]
yes it sounds like you are running pmag_gui offline and somehow your distribution is different from the current one.  are you sure you are running the latest version and that you are running it while connected to the internet?  
lisa

**Comment by hl04445567** [comment]
I'm using the same version as you showed in your screenshot (pmag_gui_4.2.23). I've also tried to re-downloaded it from the github just in case my copy was somehow compromised. And I'm sure that I'm connected to the internet. Actually, after this error occurred, I've tried to open pmag_gui on multiple computers and different internet environments, they all gave the same error.

I've attached a dataset of mine for you to test. I removed the .redo files.
[for_test.zip](https://github.com/PmagPy/PmagPy/files/3439837/for_test.zip)

Thanks for your help!

**Comment by njarboe** [comment]
I have got your data working on my computer. When you were testing did you still get the:
"-I- Couldn't connect to earthref.org, using cached method codes"
error? This might be where the problem is. Can you go to www.earthref.org in your browser when you get this error? Maybe something is blocking earthref.org access.
![Screen Shot 2019-07-29 at 10 28 24 AM](https://user-images.githubusercontent.com/1278789/62068924-040efc80-b1ec-11e9-8d28-0b7472303b88.png)

**Comment by hl04445567** [comment]
Yes, still got this error. But I can normally access www.earthref.org in browser. I've also tried to connect to a VPN server in California, still the same error. If it helps, I'm in Australia. The weiredest thing is, this error started to occurred at the same time on my school desktop, my home desktop and my laptop.

**Comment by njarboe** [comment]
This is a bit of a mystery. Do you have access to a Mac where you can install the Mac stand alone pmag_gui and try it out there?

**Comment by ltauxe** [comment]
or rather, just install the whole PmagPy distribution - not the stand alone.  The person who maintains the stand alone and who could help with this is on maternity leave. 
lisa

**Comment by hl04445567** [comment]
OK problem solved! At least for now.  I just downloaded pmag_gui.4.2.25, which is the release before the latest one. It worked fine.

The latest version is supposed to be 4.22.4 based on the title of the release, but in the actual assets the version number is 4.2.23. Perhaps a wrong version was uploaded? However, this doesn't explain why 4.2.23 was working fine until it suddenly decided to stop working one day.

Anyway, thanks for your help! Thanks for sharing this wonderful package with the community.

Re your comments above:
- @njarboe I don't have access to a Mac so unfortunately I can't test it.
- @ltauxe I have a whole PmagPy distribution installed. Tried to launch pmag_gui (performed upgrade before this) with command `python ./pmag_gui.py`, it reported the same _Couldn't connect to earthref.org_ error.
